mne/preprocessing/ica.py

```diff
@@ -270,6 +271,22 @@ def __repr__(self):
 
         return '<ICA  |  %s>' % s
 
+    def _check_for_unsupported_ica_channels(self, picks, info):
+        """Check for channels in picks that are not considered
+        valid channels. Accepted channels are the data channels
+        ('seeg','ecog','eeg','mag', and 'grad') and 'eog'.
+        This prevents the program from crashing without
+        feedback when a bad channel is provided to ICA whitening.
+        """
+        types = _DATA_CH_TYPES_SPLIT + ['eog']
+        check = all([channel_type(info, j) in types for j in picks])
```

Comment:
The set should take around the same time but make for a nicer error message. The sorting is also done for the `ValueError` message. You could put it there, but if the list is empty, the `sorted` will return almost immediately so it should be fine. And actually `sorted` constructs a list so you can even just do:
```
unknown = sorted(set(channel_type(info, pick) for pick in picks) - set(types))
```